Bring back RemoveInplaceOps()

SplitInfinity opened the pull request:
Summary: This commit brings back the `RemoveInplaceOps` pass removed in D29523283 (https://github.com/pytorch/pytorch/commit/dec5aa2260cef540b622bd9a9504b6f11cb1f607) that apparently had a bunch of internal users.

Test Plan: danthe3rd

Differential Revision: D29833316

garymm commented on torch/csrc/jit/passes/remove_inplace_ops.cpp:
```diff
@@ -34,25 +106,30 @@ void ImplicitCastForBinaryInplaceOps(Block* b) {
         (it->kind() == aten::mul_) || (it->kind() == aten::div_)) {
-      auto originalInputs = it->inputs();
-      if (originalInputs.at(0) == originalInputs.at(1)) {
+      auto orignalInputs = it->inputs();
```
please keep the fixed spelling

garymm commented on torch/csrc/jit/passes/remove_inplace_ops.cpp:
```diff
@@ -12,19 +85,18 @@ namespace jit {
-//   # Should result in a Half, but after translation to out-of-place,
-//   # would become a Float b/c Half+Float -> Float.
-//   Float : = aten::add_(%0, %1)
+// graph(%0 : Float),
```
please keep the improved comment

garymm reviewed (approved):
Please address comments before merging.

Commit:
[pytorch] Bring back RemoveInplaceOps() (#62200)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/62200

This commit brings back the `RemoveInplaceOps` pass removed in D29523283 (https://github.com/pytorch/pytorch/commit/dec5aa2260cef540b622bd9a9504b6f11cb1f607) that apparently had a bunch of internal users.

Test Plan: danthe3rd

Reviewed By: danthe3rd

Differential Revision: D29833316

fbshipit-source-id: 51f513edb021f814012d9ec28c7fe913a5d8ace5
